pkg/controller/metrics/metrics.go

```diff
@@ -655,4 +757,41 @@ func (ca *clusterAccumulator) addConditionToMap(cond hivev1.ClusterDeploymentCon
 	ca.conditions[cond][clusterType]++
 }
 
+// readMetricsConfig reads the metrics config from the file pointed to by the MetricsConfigFileEnvVar
+// environment variable.
+func readMetricsConfig() (*metricsconfig.MetricsConfig, error) {
+	path := os.Getenv(constants.MetricsConfigFileEnvVar)
+	config := &metricsconfig.MetricsConfig{}
+	if len(path) == 0 {
+		return config, nil
+	}
+
+	fileBytes, err := ioutil.ReadFile(path)
+	if err != nil || len(fileBytes) == 0 {
+		return config, err
+	}
```

Comment:
This `err` will be nil if `len(fileBytes) == 0`, causing this func to "succeed" and return an empty `MetricsConfig{}`. Is that intentional? Does the file show up as zero-length if MetricsConfig is omitted from HiveConfig? Or if it's specified but empty? If not...
```suggestion
	if err != nil {
		return config, err
	}
	if len(fileBytes) == 0 {
		return config, errors.New("empty metrics config file")
	}
```